packages/material-ui/src/TableSortLabel/TableSortLabel.d.ts

```diff
@@ -45,7 +45,7 @@ export type TableSortLabelTypeMap<
      * Sort icon to use.
      * @default ArrowDownwardIcon
      */
-    IconComponent?: React.ComponentType<{ className: string }>;
+    IconComponent?: React.ComponentType<{ className?: string }>;
```

Comment:
The difference is that the `IconComponent` you pass never receives an `undefined`  `className`. We will always inject it. `SvgIconProps` however is the type for the props you can pass in. Props that a component receives and props it actually "sees" in their render function are two different things. 

I understand that this might seem inconvenient but is not correct and can therefore cause a lot of unforeseen consequences e.g. if people passed an `IconComponent` and relied on `className` being defined they will now see type errors.

Reply:
So was the spec that I added an incorrect usage? It is highly misleading to me not to be able to pass in a component that expects `SvgIconProps`.

Reply:
To clarify our use case a bit: We have a lot of custom icons typed similarly to the spec I added. If we would change the icons type interface to **require** `className` always, it would break stuff elsewhere as more often the `className` is indeed not required (although the component does still use it, when passed) . Making a custom icon for this use case would feel a bit weird too.

Reply:
> if people passed an IconComponent and relied on className being defined they will now see type errors

@eps1lon from my perspective, we can ignore this case, it sounds like a bad practice.

Reply:
> it sounds like a bad practice

You don't declare but practices though.

> If we would change the icons type interface to require className always

It doesn't "require" it. The `className` is never nullish. So if you rely on that behavior you don't want to jump through extra type-casting shenanigans.

Reply:
@eps1lon This type interface `React.ComponentType<{ className: string }>` very much requires a component that always requires `className`. You can observe the error from the spec I added in this PR without the included type change.

Reply:
It looks like in other places (like `Button` component), the icon is typed simply as `React.ReactNode`. Using that here too would fix the issue as well.

Reply:
If there's a prop named `*Component` then `ReactNode` is likely the wrong type. Otherwise `ReactNode` is used for something completely different.

Reply:
Yeah that is true indeed. Anyways is the conclusion here that the use case specified in the spec should remain broken?